### PR TITLE
feat(map): update to v5 `map.jinja`

### PR DESCRIPTION
**Pull request by sticky-note**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

Branched from https://github.com/saltstack-formulas/consul-formula/pull/56

### Describe the changes you're proposing

update to v5 `map.jinja`

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by sticky-note** [comment]
@myii : I've just rebased this with new master `v1.0.2`
I am not sure this is a breaking change

File: consul/config.sls [hunk 1]
```diff
@@ -25,7 +29,12 @@ consul-script-install-{{ loop.index }}:
     - template: jinja
     - context: {{ script.get('context', {}) | yaml }}
     - user: {{ consul.user }}
-    - group: {{ consul.group }}
+    - group: {{ consul.group }}# -*- coding: utf-8 -*-
+# vim: ft=sls
+---
+{#- Get the `tplroot` from `tpldir` #}
+{%- set tplroot = tpldir.split("/")[0] %}
+{%- from tplroot ~ "/map.jinja" import mapdata with context %}
```
Comment (myii) [inline consul/config.sls]: Looks like this was pasted by mistake.

**Review by myii — CHANGES_REQUESTED** [review]
Only a quick review for now but there is one glaring problem, as mentioned inline.

Otherwise, the `map.jinja` verifier has really helped a lot to confirm that the v5 update appears to have been implemented correctly!

File: consul/config.sls (hunk 1 above)
Comment (sticky-note) [reply consul/config.sls]: oups, sorry for the paste mistake  :-1:

Comment (sticky-note) [reply consul/config.sls]: @myii Updated the request

**Comment by sticky-note** [comment]
@myii PR seems to be ok now

File: consul/libtofs.jinja [hunk 2]
```diff
@@ -0,0 +1,115 @@
+{%- macro files_switch(
```
Comment (myii) [inline consul/libtofs.jinja]: This macro is unused for this particular PR, so please remove this file for now (and add it in #58).  Also, `docs/TOFS_pattern.rst` should be added in #58 as well.

**Review by myii — CHANGES_REQUESTED** [review]
Almost there, @sticky-note.  One change mentioned inline (below) and one more here.

https://github.com/saltstack-formulas/consul-formula/blob/ce35c1ca66b015190a8d671e8dd0e56faeed97c5/CODEOWNERS#L21-L26

This needs to be adjusted in this way;

```diff
 /docs/TOFS_pattern.rst                      @saltstack-formulas/ssf
 /*/_mapdata/                                @saltstack-formulas/ssf
+/*/libmapstack.jinja                        @saltstack-formulas/ssf
+/*/libmatchers.jinja                        @saltstack-formulas/ssf
 /*/libsaltcli.jinja                         @saltstack-formulas/ssf
 /*/libtofs.jinja                            @saltstack-formulas/ssf
+/*/map.jinja                                @saltstack-formulas/ssf
 /test/integration/**/_mapdata.rb            @saltstack-formulas/ssf
 /test/integration/**/libraries/system.rb    @saltstack-formulas/ssf
```

---

Reviewed using https://github.com/myii/ssf-formula/pull/434.

**Comment by sticky-note** [comment]
@myii Requested changes are done.

**Review by myii — APPROVED** [review]

**Comment by myii** [comment]
Fantastic work, @sticky-note -- merged, thanks for the patience.

**Pull request merged**